articles/per_package_documentation.md

```diff
@@ -0,0 +1,112 @@
+That project, in the implementation phase now, includes some keys points that affect the context of the per-package documentation.
+
+- All documentation will be hosted on `docs.ros.org`
+- All documentation will be versioned by ROS 2 distribution names
+- The URL structure will be `docs.ros.org/<lang>/<distro>/<page>` for generic documentation and `docs.ros.org/<lang>/<distro>/p/<package_name>/<page>` for package documentation
+  - For more context on where documentation will fall under this URL structure, see [this diagram](https://docs.google.com/drawings/d/1KxzDrcSZzwGgudk-kEXGWnoRuAtc1ffl_KBeIu-V60Y/edit?usp=sharing).
+
+## Requirements
+
+The "Primary requirements" are those that must be in place for the system to be functional and achieve its purpose.
+The "Secondary requirements" are important features and functionality that are not necessary to roll out the first stage of implementation, but should be carried out following the system's roll out.
+
+### Primary requirements
+
+**All package documentation must be available with the rest of the ROS 2 documentation under a single domain**
+
+Package documentation will be treated the same as the generic ROS 2 documentation.
+Its presence as part of the docs site should be made well-known from the overview as well as any entry point into the docs site, and be intuitive from the organization and layout of the site.
+
+**Package documentation must have an easily navigable and intuitive URL structure**
+
+Every package's "home page" should be reachable with minimal effort by the scheme `docs.ros.org/p/<package_name>/`, which will redirect to default language and version, `docs.ros.org/<lang>/<distro>/p/<package_name>/`  
+
+**Package documentation must be maintainable in its repository without going through a third party**
+
+Maintainers and contributors will only have to work on their package's documentation within its repository.
+The details of building and hosting will not be a concern of package maintainers.
+
+Despite being hosted alongside the generic documentation, working on a package's documentation will not require any work on the repositories of the generic documentation (currently `ros2/ros2_documentation`) or the site repository (currently `ros-infrastructure/rosindex`).
+
+**The system must support C++ and Python API docs generation**
+
+The system will automatically extract API docs for these languages from the source code in package repositories and build the output on the docs site.
+
+**The system must allow package documentation to be versioned per ROS 2 distribution**
+
+The docs site and buildfarm will allow documentation for the latest version of a package corresponding to each ROS 2 distribution the docs site supports versioning for.
+This means no versioning for several versions of a package per ROS 2 distribution.
+The package repository can still maintain it's own docs for previous versions, they just won't be hosted on our site.
```

Comment:
I think we said here we wanted the URL structure to support this.  So I think we should clarify this a bit and say that while multiple versions isn't currently supported, the URL structure will support it and it is on the secondary requirements board below.